contexts/SubscriptionContext/tests/Integration/UseCases/AddSubscription/AddSubscriptionUseCaseTest.php

```diff
@@ -139,4 +141,19 @@ public function testGivenDataThatNeedsToBeModerated_requestNotBeMailed() {
 		$useCase->addSubscription( $request );
 	}
 
+	public function testNewMailAddressFromSubscription(): void {
+		$useCase = new AddSubscriptionUseCase( $this->repo, $this->validator, $this->mailer );
+		$class = new ReflectionClass( AddSubscriptionUseCase::class );
```

Comment:
Sneaking... The alternative being to insist on not using reflection, claiming that only public API needs testing, and never writing this test.
I removed a constructor parameter and no test twitched an eye, so I wanted to at least make sure that the flow of the relevant data (the mail address) is tested in preparation for the refactoring due in T166687 - fully aware that it's not beautiful, but working with what we got. After #896 we've got none of this security.